### ISSUE TITLE
Convert forms to Redux Form

### ISSUE THREAD

**Issue by sogasg** [dOrgTech/DAOcreator#133]
https://github.com/erikras/redux-form

**Comment by sethfork** [comment]
Global redux stores aren't the best fit for form state - the general consensus is to use local/component state. There are libraries that make working with stateful forms simpler, such as [React Final Form](https://github.com/final-form/react-final-form), which is what I would propose using.

**Comment by dOrgJelli** [comment]
Form refactoring is currently underway in the branch "deployment-soft-fixes" (strange name, I know... I got side tracked).

I've chosen to not use redux forms or formik because I want to be able to reuse these forms outside of react. For example if we wanted to make a CLI. To see an example use-case, see here (still very WIP):
https://github.com/dOrgTech/DAOcreator/blob/3305ab8d9112c25a56898dd766ad98b551ba0cd7/src/components/pages/DAOcreator/NamingStep.tsx#L54

The React specific drawers lives here:
https://github.com/dOrgTech/DAOcreator/blob/deployment-soft-fixes/src/components/common/FormField.tsx

And the implementation lives here:
https://github.com/dOrgTech/DAOcreator/tree/deployment-soft-fixes/src/lib/forms

I made sure that type correctness is maintained throughout, so if you mistype a form name (ex: "reputationn"), a compiler error will be thrown.
